File: src/main/java/net/frozenorb/potpvp/util/nametag/construct/NameTagUpdate.java
package net.frozenorb.potpvp.util.nametag.construct;

import lombok.Getter;
import org.bukkit.entity.Player;

@Getter
public class NameTagUpdate {

    private String toRefresh;
    private String refreshFor;

    public NameTagUpdate(Player toRefresh) {
        if(toRefresh == null) return;

        this.toRefresh = toRefresh.getName();
    }

    public NameTagUpdate(Player toRefresh, Player refreshFor) {
        this.toRefresh = toRefresh.getName();
        this.refreshFor = refreshFor.getName();
    }
}

File: src/main/kotlin/net/frozenorb/potpvp/kt/nametag/NametagUpdate.kt
package net.frozenorb.potpvp.kt.nametag

import org.bukkit.entity.Player

internal class NametagUpdate(var toRefresh: String, var refreshFor: String?) {
    constructor(toRefresh: Player): this(toRefresh.name, null)
    constructor(toRefresh: Player, refreshFor: Player): this(toRefresh.name, refreshFor.name)
}
